### ISSUE TITLE
Stop modify serializer options

### ISSUE THREAD

**Issue by sungam3r** [graphql-dotnet/graphql-dotnet#2865]
See https://github.com/graphql-dotnet/graphql-dotnet/pull/2859/files#r786709872

**Comment by sungam3r** [comment]
I ran into this porting my code on v5.

**Comment by Shane32** [comment]
@sungam3r The above PRs were merged -- is this complete?

**Comment by sungam3r** [comment]
Nope. Search for `// TODO: fix this: it modifies serializerOptions`.
